## [1629] 곱셈

In [5]:
import sys
input = sys.stdin.readline

a, b, c = map(int, input().split())

def mul(a, y):
    if y == 1:
        return a % c
    else:
        tmp = mul(a, y//2)
        
        if y % 2 == 0:
            return (tmp * tmp) % c # 짝수 
        else:
            return (tmp * tmp * a) % c # 홀수

print(mul(a, b))

10 11 12
4


어렵다 !!! 시간 초과가 안 나게 하고 재귀 문제가 생기지 않도록 어떻게 풀 수 있을까 계속 고민했는데 잘 안 떠올랐다..! ㅠㅠ <br>

**문제 풀이** <br>
만약 10, 11, 12가 a, b, c에 들어가게 된다면, `10**11 % 12`가 계산되어야 한다. <br>
이걸 문제에서는 나눠서 `[(10**5) * (10**5) * 10] % 12`로 계산하는 방식이다. <br>
그래서 몇 번 나누는지에 대한 변수 b가 짝수라면, `*a`를 따로 해주지 않고, 홀수면 한 번 더 곱해준다.